### PR TITLE
[bug 2340] first part of "buildslave" to "buildbot-worker" transition

### PR DESCRIPTION
**Pull request by rutsky**
This is  #2031 rebased against bug2340 branch with [coping "slave" directory to "worker" commit](https://github.com/buildbot/buildbot/commit/d6eaa56cf130666bbb8995d7dd86ac01073b04b8).

Changes:

1. `buildslave` package was renamed to `buildbot-worker` (including all references to it);
2. `buildslave` script was renamed to `buildbot-worker`;
3. update scripts in `contrib` (however I didn't tested them);
4. update man page in `docs`.

Now there should be fully functional `buildbot_worker` package and `buildbot-worker` runner script (i.e. all tests should pass), but with "slave" stuff inside.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -21,12 +21,12 @@ install:
...
   - "copy /y sqlite3.dll %PYTHON%\\DLLs\\"
   
-  - "%PYTHON%\\python.exe -m pip install -e pkg -e master[test] -e slave"
+  - "%PYTHON%\\python.exe -m pip install -e pkg -e master[test] -e slave -e worker"
 
 build: false
 
 test_script:
-  - "%PYTHON%\\python.exe %PYTHON%\\Scripts\\trial.py --reporter=text --rterrors buildbot.test buildslave.test"
+  - "%PYTHON%\\python.exe %PYTHON%\\Scripts\\trial.py --reporter=text --rterrors buildbot.test buildslave.test buildbot_worker.test"
```
Comment (tardyp) [inline appveyor.yml]: this will clash with the appweyor PR I just merged

Comment (rutsky) [inline appveyor.yml]: Yep, I'll rebase/merge this and bug2340 branch later. Also this branch will conflict with #2030.

File: worker/contrib/os-x/net.sourceforge.buildbot.worker.plist [hunk 2]
```diff
@@ -4,7 +4,7 @@
 <plist version="1.0">
 <dict>
          <key>Label</key>
-         <string>net.sourceforge.buildbot.slave</string>
+         <string>net.sourceforge.buildbot.worker</string>
```
Comment (tardyp) [inline worker/contrib/os-x/net.sourceforge.buildbot.worker.plist]: sourceforge??

File: worker/contrib/windows/buildbot_service.py [hunk 3]
```diff
@@ -104,7 +104,7 @@
 class BBService(win32serviceutil.ServiceFramework):
     _svc_name_ = 'BuildBot'
     _svc_display_name_ = _svc_name_
-    _svc_description_ = 'Manages local buildbot slaves and masters - ' \
+    _svc_description_ = 'Manages local buildbot workers and masters - ' \
                         'see http://buildbot.sourceforge.net'
```
Comment (tardyp) [inline worker/contrib/windows/buildbot_service.py]: sourceforge?

Comment (rutsky) [inline worker/contrib/windows/buildbot_service.py]: Yes, I noticed this too :)
This is on my TODO list (this is not the only place where SourceForge is in codebase), lets fix this later.

**Comment by tardyp** [comment]
LGTM..appart from some nits.

Damn there are pretty old stuff there. I bet half of them do not work any more.

**Comment by tardyp** [comment]
ok, I'll let it there for a couple of day to let others have a look (@djmitche ? @jaredgrubb ? @sa2ajj ?)to have a look

File: worker/contrib/os-x/net.sourceforge.buildbot.worker.plist (hunk 2 above)
Comment (djmitche) [inline worker/contrib/os-x/net.sourceforge.buildbot.worker.plist]: Plists just require a reversed DNS name to make a unique label.  This was form back when we were on SF.

File: worker/contrib/windows/buildbot_service.py (hunk 3 above)
Comment (djmitche) [inline worker/contrib/windows/buildbot_service.py]: FWIW I'm not *terribly* worried about anything in `contrib/`.  It's fine to fix things as you see them, but don't put too much effort into it.

**Comment by djmitche** [comment]
LGTM

We don't get so many patches to the `slave/` subdirectory that we need to institute any kind of change freeze, but we should all bear in mind that such patches will need to be copied over to `worker/` for the duration of both directories existing.

**Pull request merged**